src/vmod_soap.c

```diff
@@ -172,8 +172,10 @@ int process_request(struct priv_soap_task *task, enum soap_state state)
 				VSLb(task->ctx->vsl, SLT_Error, "Not enough data");
 				return (-1);
 			}
-			while (task->bytes_left > 0) {
-				int bytes_read = read_body_part(task->req_http, task->bytes_left);
+			int bytes_total = task->bytes_left;
```

Comment:
`bytes_left` is a `ssize_t` but you use a `int`. It should not an issue in most use-cases, but we have to preserve the type as much as we can (ditto for `bytes_read` IMHO)